### ISSUE TITLE
Independent communication from the isolate to main thread

### ISSUE THREAD

**Issue by synw** [renatoathaydes/actors#3]
Hi. Would it be possible to implement independent communication from the isolate to the main thread? Actually the api works like the Flutter `compute` function and awaits a result from something sent from the main thread: the isolate can not send data to the main thread by itself.

My feature request is to have some kind of listener in the main thread that can receive the data from the isolate at any moment and independently of any call. This way the isolate can send data at any time and not only when the main thread requests something. I wrote a library that does that: [Iso](https://github.com/synw/iso) but I would prefer using your api as I find it superior in terms of readability and ease of use.

**Comment by renatoathaydes** [comment]
Hi @synw !

My idea to support this use case is to simply allow an actor to return a `Stream`, so you could do this easily:

```dart
actor.send(#listen).listen((data) { 
  // handle data
});
```

What do you think?

**Comment by synw** [comment]
Yes it would be good. About the api I would prefer the stream to have it's own name instead of chaining from `send`: something like that:

```dart
actor.data.listen((dynamic data) { 
  // handle data
});
```
Another suggestion: it would be great to be able to use typed data as an option:

```dart
StreamSubscription<Map<String, dynamic>> sub;

sub = actor.data.listen((Map<String, dynamic> data) { 
  // handle data
});
```
If the type is not declared it could still use dynamic data by default. It would add safety and avoid having to do type casts after receiving some data

**Comment by renatoathaydes** [comment]
Just released version 0.5.0 of actors.
There's a new `StreamActor` which is basically an `Actor<M, Stream<A>>` (i.e. an actor that receives messages of type `M` and returns a `Stream<A>`).
This allows you to have messages sent from a different Isolate asynchronously.

> Another suggestion: it would be great to be able to use typed data as an option...

Actors are typed. If you want to receive messages of type `Map<String, dynamic>>`, then you'll just use a `StreamActor<M, Map<String, dynamic>>`, then listen:

```dart
final actor = StreamActor<String, Map<String, dynamic>>(handler);
await for (final message in actor.send('give me data')) {
    final name = message['name'];
    print("Got a $name");
}
```

Hope this lets you implement what you had in mind.
